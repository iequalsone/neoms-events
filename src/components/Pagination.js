import React, { Component } from "react";

class Pagination extends Component {
  renderPrevLink(currentPage, totalPages) {
    let prevPage = 1;
    // console.log(currentPage);

    if (typeof currentPage === "undefined" || currentPage <= 1) {
      return;
    }

    if (currentPage !== 1 && currentPage <= totalPages) {
      prevPage = --currentPage;
    }

    if (currentPage > totalPages) {
      prevPage = --currentPage;
    }

    return (
      <a href={"/" + process.env.REACT_APP_BASE_URL + "/" + prevPage}>
        Previous
      </a>
    );
  }

  renderNextLink(currentPage, totalPages) {
    let nextPage = 2;

    if (currentPage >= totalPages) {
      return;
    }

    if (currentPage <= totalPages) {
      nextPage = ++currentPage;
    }

    return (
      <a href={"/" + process.env.REACT_APP_BASE_URL + "/" + nextPage}>Next</a>
    );
  }

  render() {
    const { currentPage, itemsPerPage, totalItems } = this.props;
    const totalPages = totalItems / itemsPerPage;
    return (
      <div className="row event-pagination">
        <div className="col-xs-6">
          <p className="text-right">
            {this.renderPrevLink(currentPage, totalPages)}
          </p>
        </div>
        <div className="col-xs-6">
          <p className="text-left">
            {this.renderNextLink(currentPage, totalPages)}
          </p>
        </div>
      </div>
    );
  }
}

export default Pagination;
