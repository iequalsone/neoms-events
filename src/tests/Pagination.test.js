import React from 'react';
import Pagination from '../components/Pagination';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const component = shallow(
    <Pagination />
  );
  expect(component).toMatchSnapshot();
});