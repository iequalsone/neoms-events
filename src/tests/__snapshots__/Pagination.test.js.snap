// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-6"
>
          <p
                    className="text-right"
          />
</div>,
        <div
          className="col-xs-6"
>
          <p
                    className="text-left"
          >
                    <a
                              href="/events/2"
                    >
                              Next
                    </a>
          </p>
</div>,
      ],
      "className": "row event-pagination",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="text-right"
/>,
          "className": "col-xs-6",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "text-right",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="text-left"
>
            <a
                        href="/events/2"
            >
                        Next
            </a>
</p>,
          "className": "col-xs-6",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/events/2"
>
              Next
</a>,
            "className": "text-left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "href": "/events/2",
            },
            "ref": null,
            "rendered": "Next",
            "type": "a",
          },
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-6"
>
            <p
                        className="text-right"
            />
</div>,
          <div
            className="col-xs-6"
>
            <p
                        className="text-left"
            >
                        <a
                                    href="/events/2"
                        >
                                    Next
                        </a>
            </p>
</div>,
        ],
        "className": "row event-pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="text-right"
/>,
            "className": "col-xs-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "text-right",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="text-left"
>
              <a
                            href="/events/2"
              >
                            Next
              </a>
</p>,
            "className": "col-xs-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="/events/2"
>
                Next
</a>,
              "className": "text-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "href": "/events/2",
              },
              "ref": null,
              "rendered": "Next",
              "type": "a",
            },
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
